### DISCUSSION TITLE
What happens if all tokio tasks fall into sleep?

### DISCUSSION THREAD

**Discussion by wqshr12345** [tokio-rs/tokio#5940] [Q&A] [answered]
Hello, I encountered an issue while using Tokio. My code is shown below:
```
#[tokio::test]
async fn tests() {
    println!("Hello, world!");
    tokio::spawn(async {
        sleep(Duration::from_secs(1)).await;
        println!("Hello from task2!");
    });
    sleep(Duration::from_secs(10)).await;
    println!("Hello from task1!");
}
```
The expected output  is as follows:
```
Hello, world!
Hello from task2!
Hello from task1!
```
However, the actual output I'm getting is:
```
Hello, world!
Hello from task1!
Hello from task2!
```
I believe that task2 should wake up from the sleep first and execute before anything else, but the actual code behavior is different from what I expected. Can anyone help me with an explanation?

**Comment by Darksonn** [comment]
The code you posted will output the correct output. My guess is that the code you posted and the code you actually ran is different. Specifically, I'm guessing that the code you ran used `std::thread::sleep` instead (which does not need an await). If that's the case, then the problem is explained by [this article](https://ryhl.io/blog/async-what-is-blocking/).

**Reply by wqshr12345** [reply]
I might have missed a bit.I'm using tokio_test::task::spawn. 
On my computer, it seems that the program prints "Hello, world! Hello from task1!" and then exits.

**Reply by Darksonn** [reply, marked as answer]
The `tokio_test::task::spawn` utility doesn't do the same thing as `tokio::spawn`. It's intended for the use-case where you want to control when the task should be polled from the test. Since you never do that, it doesn't get polled.
